homeassistant/components/melcloud/sensor.py

```diff
@@ -170,28 +171,23 @@ def __init__(
         self._api = api
         self.entity_description = description
 
-        self._attr_name = f"{api.name} {description.name}"
         self._attr_unique_id = f"{api.device.serial}-{api.device.mac}-{description.key}"
+        self._attr_device_info = api.device_info
 
         if description.device_class == SensorDeviceClass.ENERGY:
             self._attr_state_class = SensorStateClass.TOTAL_INCREASING
         else:
             self._attr_state_class = SensorStateClass.MEASUREMENT
```

Comment:
For a follow up PR, you can set this in the entity description as well :)

Reply:
Do you want that I move this to entity description here or prefer to have a follow up PR?